recipes/default.rb

```diff
@@ -14,44 +14,46 @@
 nnPort=node['hops']['nn']['port']
 hops_group=node['hops']['group']
 my_ip = my_private_ip()
-my_public_ip = my_public_ip()
-rm_private_ip = private_recipe_ip("hops","rm")
-zk_ip = private_recipe_ip('kzookeeper', 'default')
-
-# Convert all private_ips to their hostnames
-# Hadoop requires fqdns to work - won't work with IPs
-hostf = Resolv::Hosts.new
 
 ndb_connectstring()
 
+## Do not try to discover Hopsworks before it has been actual deployed
+## default recipe is included by hops::ndb
+run_list = node.primary_runlist
+run_discovery_recipes = ['recipe[hops::client]', 'recipe[hops::dn]', 'recipe[hops::jhs]', 'recipe[hops::nm]', 'recipe[hops::nn]', 'recipe[hops::ps]', 'recipe[hops::rm]', 'recipe[hops::rt]']
+run_discovery = false
+for dr in run_discovery_recipes do
+  if run_list.include?(dr)
+    run_discovery = true
+    break
+  end
+end
+
+hopsworks_port = ""
+if run_discovery
+  ruby_block 'Discover Hopsworks port' do
+    block do
+      _, hopsworks_port = consul_helper.get_service("glassfish", ["http", "hopsworks"])
+      if hopsworks_port.nil?
+        raise "Could not get Hopsworks port from local Consul agent. Verify Hopsworks is running with service name: glassfish and tags: [http, hopsworks]"
+      end
+    end
+  end
+end
+
+glassfish_fqdn = consul_helper.get_service_fqdn("glassfish")
+
 rpcSocketFactory = "org.apache.hadoop.net.StandardSocketFactory"
 hopsworks_crl_uri = "RPC TLS NOT ENABLED"
 if node['hops']['tls']['enabled'].eql? "true"
   rpcSocketFactory = node['hops']['hadoop']['rpc']['socket']['factory']
-  hopsworks_crl_uri = "#{hopsworks_host()}#{node['hops']['tls']['crl_fetch_path']}"
 end
 
 node.override['hops']['hadoop']['rpc']['socket']['factory'] = rpcSocketFactory
 
-if node['hops']['nn']['private_ips'].length > 1
-  allNNIps = node['hops']['nn']['private_ips'].join(":#{nnPort},") + ":#{nnPort}"
-else
-  allNNIps = "#{node['hops']['nn']['private_ips'][0]}" + ":#{nnPort}"
-end
-
-# This is a namenode machine, the rpc-address in hdfs-site.xml is used as "bind to" address
-if node['hops']['nn']['private_ips'].include?(my_ip)
-  nn_rpc_address = "#{my_ip}:#{nnPort}"
-  nn_http_address = "#{my_ip}:#{node['hops']['nn']['http_port']}"
-  nn_https_address = "#{my_ip}:#{node['hops']['dfs']['https']['port']}"
-else
-  # This is a non namenode machine, a random namenode works
-  nn_rpc_address = private_recipe_ip("hops", "nn") + ":#{nnPort}"
-  nn_http_address = private_recipe_ip("hops", "nn") + ":#{node['hops']['nn']['http_port']}"
-  nn_https_address = private_recipe_ip("hops", "nn") + ":#{node['hops']['dfs']['https']['port']}"
```

Comment:
@kouzant do we need to modify this module as a result of this PR? https://github.com/logicalclocks/hops-util-py/blob/branch-1.2/hops/hdfs.py#L843

Reply:
Yes, we should query service discovery to get the port

Reply:
Thanks, so we need a hops-util-py PR as part of this JIRA as well.

Reply:
I opened a PR for this https://github.com/logicalclocks/hops-util-py/pull/158